File: Data Structures & Algorithms/reverse-string/submission-1.java
class Solution {
    public void reverseString(char[] s) {
        int left = 0;
        int right = s.length - 1;

        while (left < right) {
            char stored = s[right];
            s[right] = s[left];
            s[left] = stored;

            left++;
            right--;
        }
    }
}

File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun invertTree(root: TreeNode?): TreeNode? {
        if (root == null) return null

        root.left = root.right.also { root.right = root.left }

        invertTree(root.right)
        invertTree(root.left)

        return root
    }
}
